File: Java/Algorithms/linsearch.java
import java.util.Scanner;

public class linsearch {

    public static void main(String[] args) {
        Scanner in = new Scanner(System.in);
        System.out.println("Enter the size of the array: ");
        int n = in.nextInt();
        int[] arr = new int[n];
        System.out.println("Enter the elements: ");
        for(int i=0; i<arr.length; i++)
        {
            arr[i] = in.nextInt();
        }
        System.out.println("Enter the element to be searched: ");
        int key = in.nextInt();
        int index = lsearch(arr, key);
        if(index==-1)
        {
            System.out.println("Key is not present in the array.");
        }
        else
            {
                System.out.println("The key is present at position " + (index+1));
            }
        in.close();
    }

    static int lsearch(int[] arr, int key)
    {
        int i;
        for(i=0; i<arr.length; i++)
        {
            System.out.println("h1");
            if(arr[i]==key)
            {
                System.out.println("h2");
                return i;
            }
        }
        return -1;
    }
    
}

File: C/3rd Sem/DSA/lin_search.c
#include<stdio.h>
void read(int n,int a[]);
int lin_search(int n, int a[], int key);
int main()
{
    int n, a[100], x, key;
    printf("Enter the number of elements:\n");
    scanf("%d",&n);
    printf("Enter the elements:\n");
    read(n, a);
    printf("Enter the element to be searched:\n");
    scanf("%d",&key);
    x=lin_search(n,a,key);
    if(x==-1)
        printf("The element is not present");
        else 
            printf("The element is present at %d",x);

}

void read(int n, int a[])
{
    int i;
    for(i=0;i<n;i++)
        scanf("%d",&a[i]);
}

int lin_search(int n, int a[], int key)
{
    int i;
    for(i=0;i<n;i++)
    {
        if(key==a[i])
        {
            return i;
        }
    }
    return -1;
}